### DISCUSSION TITLE
R-29 "Big Rig" airlock doors making unnecessary noise

### DISCUSSION THREAD

**Discussion by RiceOnDry** [FakeFishGames/Barotrauma#14648] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

I noticed that both of R-29's outer airlock doors make noise as if water was flowing in even though the airlock is full of water. Not a game breaker but definitely annoying.

### Reproduction steps

1. Get a R-29
2. Start an expedition
3. Open either the upper or lower airlock outer door and keep it open
4. Airlock fills with water
5. Door keeps making noise

Relevant video:
https://github.com/user-attachments/assets/f4a3798e-1834-4395-ab9b-aacd03f55f6c

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Single player

### Version

v1.5.9.2 (Summer Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by SAYtan47** [comment]
Im sure it works as it should be, its just the way how airlocks are wired. Theres a pump and a water detector in the airlock. The moment water detector senses water the pump starts pumping, creating this noise. Pump doesnt account for the state of the hatch itself, so it goes on pumping even while hatch is open and water is flowing inside. Same happens for multiple subs actually. This noise out of nowhere is just pump trying to dry the open airlock.

**Reply by RiceOnDry** [reply]
Oh, I now see what's happening. Realistically it makes sense but it's just the way the game's coded. Also, it might be possible but risky to wire the pump in a way that it won't try to pump out water while the door is open, I'm assuming?

**Reply by SAYtan47** [reply]
Sorry for being late, but yeah, you can add additional check to the pump if airlock is closed or not. Im not sure what drawbacks in terms of safety this will cause, but nothing lethal for sure (tell me if you die because of it)

**Comment by Regalis11** [comment]
This has been fixed in our private repo now, fix coming up in a future update!
